From e9904b4da85ecb5c22294a1439dd3192a90b9680 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 1 May 2020 16:25:40 +0200
Subject: [PATCH 0097/1695] core: mobj: initialize earlier generic memory
 object

Change generic mobj support to be initialized from the service
initialization level instead of driver and driver_late initialization
levels.

This change allows driver to benefit from mobj abstraction
during their initialization.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e653976b0a038fa96f8f86bf42f26da013222971 core: mobj: initialize earlier generic memory object]
Change-Id: I1d64e043cc9c74b79da8bf47666e559b5cf0ce8a
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: picked from rebase-3.19 development branch]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/mm/mobj.c       | 2 +-
 core/tee/entry_std.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/mm/mobj.c b/core/mm/mobj.c
index afc8176d1..b22e9c1c9 100644
--- a/core/mm/mobj.c
+++ b/core/mm/mobj.c
@@ -540,4 +540,4 @@ static TEE_Result mobj_init(void)
 	return TEE_SUCCESS;
 }
 
-driver_init_late(mobj_init);
+service_init(mobj_init);
diff --git a/core/tee/entry_std.c b/core/tee/entry_std.c
index d3ae62486..7f6fa433d 100644
--- a/core/tee/entry_std.c
+++ b/core/tee/entry_std.c
@@ -606,4 +606,4 @@ static TEE_Result default_mobj_init(void)
 	return TEE_SUCCESS;
 }
 
-driver_init_late(default_mobj_init);
+service_init(default_mobj_init);
-- 
2.47.2

